tests/test_integration.py

```diff
@@ -57,5 +57,5 @@ def test_search_dist_dirs(self):
         distribution metadata dirs. Protect it for PyPA
         use-cases (only). Ref python/importlib_metadata#111.
         """
-        res = MetadataPathFinder._search_paths('any-name', [])
+        res = MetadataPathFinder()._search_paths('any-name', [])
```

Comment:
This change invalidates a use-case that it was intended to protect.

Reply:
I can move the cache to the class (instead of it being per-instance, but the effect would more or less turn MetadataPathFinder into a singleton class (because everything would be in class vars instead of instance vars).  Would that work for you?